Example-Swift/PhotoViewerCoordinator.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  PhotoViewerCoordinator.swift
+//  NYTPhotoViewer
+//
+//  Created by Chris Dzombak on 2/2/17.
+//  Copyright © 2017 NYTimes. All rights reserved.
+//
+
+import NYTPhotoViewer
+
+/// Coordinates interaction between the application's data layer and the photo viewer component.
+final class PhotoViewerCoordinator: NYTPhotoViewerDataSource {
+    let slideshow: [NYTPhotoBox]
+    let provider: PhotosProvider
+
+    lazy var photoViewer: NYTPhotosViewController = {
+        return NYTPhotosViewController(dataSource: self)
+    }()
+
+    init(provider: PhotosProvider) {
+        self.provider = provider
+        self.slideshow = provider.fetchDemoSlideshow().map { NYTPhotoBox($0) }
+        self.fetchPhotos()
+    }
+
+    func fetchPhotos() {
+        for box in slideshow {
+            provider.fetchPhoto(named: box.value.name, then: { [weak self] (result) in
+                box.image = result
+                self?.photoViewer.update(box)
+            })
+        }
+    }
+
+    // MARK: NYTPhotoViewerDataSource
+
+    @objc
+    var numberOfPhotos: NSNumber? {
```

Comment:
On Xcode 8.2.1, this leaves me with an error:

<img width="484" alt="screen shot 2017-02-15 at 16 55 45" src="https://cloud.githubusercontent.com/assets/102904/22997268/cb6fde24-f39f-11e6-821b-5879c58b5946.png">